File: apps/mobile/lib/features/orders/domain/entities/order.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'order.freezed.dart';
part 'order.g.dart';

enum OrderStatus { pending, confirmed, preparing, delivering, delivered, cancelled }

@freezed
abstract class Order with _$Order {
  const factory Order({
    required String id,
    required String userId,
    required String stallId,
    required List<Map<String, dynamic>> items,
    required double totalAmount,
    required String seatZone,
    required String seatNumber,
    @Default(OrderStatus.pending) OrderStatus status,
    String? paymentId,
    String? trackingWsUrl,
    DateTime? createdAt,
    DateTime? deliveredAt,
  }) = _Order;

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
}

@freezed
abstract class QueueInfo with _$QueueInfo {
  const factory QueueInfo({
    required String stallId,
    required String stallName,
    required double waitMins,
    required int queueLength,
    required double forecast15min,
    required String source,
    @Default([]) List<String> alternativeStallIds,
  }) = _QueueInfo;

  factory QueueInfo.fromJson(Map<String, dynamic> json) =>
      _$QueueInfoFromJson(json);
}

File: apps/api/app/models/order.py
from pydantic import BaseModel, Field
from typing import Optional, Literal
from datetime import datetime, timezone
import uuid


class Order(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    user_id: str
    stall_id: str
    items: list[dict]
    total_amount: float
    seat_zone: str
    seat_number: str
    status: Literal[
        "pending", "confirmed", "preparing", "delivering", "delivered", "cancelled"
    ] = "pending"
    payment_id: Optional[str] = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    delivered_at: Optional[datetime] = None


class OrderCreate(BaseModel):
    user_id: str
    stall_id: str
    items: list[dict]
    seat_zone: str
    seat_number: str


class QueueSnapshot(BaseModel):
    stall_id: str
    wait_mins: float
    queue_length: int
    forecast_15min: float
    source: Literal["sensor", "crowd", "pos"]
    alternative_stall_ids: list[str] = Field(default_factory=list)
    recorded_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
